gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -320,14 +368,13 @@ Filter constructFilterFromChildNodesResult(EntityFetcherResponse result) {
   }
 
   @Override
-  public EntityFetcherResponse visit(SortAndPaginateNode sortAndPaginateNode) {
-    EntityFetcherResponse result = sortAndPaginateNode.getChildNode().acceptVisitor(this);
-    // Set the total entities count in the execution context before pagination
-    executionContext.setTotal(result.size());
```

Comment:
Isn't this still needed? What if there is no DataFetcherNode at all. Who sets this in such a case?

Reply:
Good catch. The current scenarios always had a DataFetcherNode, but yes, I will add it here as well. That will account for the optimization here https://github.com/hypertrace/gateway-service/pull/61#discussion_r556570387